.github/workflows/ci-flutter-main.yml

```diff
@@ -0,0 +1,33 @@
+# The `check-flutter-main` job serves 2 purposes:
+# 1. It checks that the code works with the flutter's latest main
+#    channel.
+# 2. It checks that our code environment setup works with a typical
+#    Flutter SDK installation setup (via ~/flutter).
```

Comment:
I started writing a comment saying I wasn't convinced this second point was really testing anything, but then decided it is after all. I don't think "our code environment setup" quite captures it, though — that sounds to me like things like `tools/setup-vendor-flutter` that are precisely what isn't getting exercised here. So here's a rewording:

```suggestion
# 2. It checks that tools/check and our related scripts don't embed
#    any hidden assumptions that the Flutter SDK lives at vendor/flutter/.
```